UdpConnectedIntegrationSpec: must be able to send and receive when server goes away (and comes back)

https://github.com/akka/akka/runs/3239219181?check_suite_focus=true

```
must be able to send and receive when server goes away (and comes back) *** FAILED *** (1 second, 44 milliseconds)
[info]   java.lang.AssertionError: assertion failed: expected 3, found Received(ByteString(68, 97, 116, 97, 32, 116, 111, 32, 110, 101, 119, 32, 105, 110, 99, 97, 114, 110, 97, 116, 105, 111, 110),/127.20.222.105:46221)
[info]   at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
[info]   at akka.testkit.TestKitBase.expectMsg_internal(TestKit.scala:462)
[info]   at akka.testkit.TestKitBase.expectMsg(TestKit.scala:438)
[info]   at akka.testkit.TestKitBase.expectMsg$(TestKit.scala:159)
[info]   at akka.testkit.TestKit.expectMsg(TestKit.scala:973)
[info]   at akka.io.UdpConnectedIntegrationSpec.f$proxy6$1(UdpConnectedIntegrationSpec.scala:156)
[info]   at akka.io.UdpConnectedIntegrationSpec.$init$$$anonfun$7$$anonfun$6(UdpConnectedIntegrationSpec.scala:128)
```

--- a/akka-actor-tests/src/test/scala/akka/io/UdpConnectedIntegrationSpec.scala
+++ b/akka-actor-tests/src/test/scala/akka/io/UdpConnectedIntegrationSpec.scala
@@ -10,7 +10,6 @@ import scala.concurrent.duration._
 
 import akka.actor.ActorRef
 import akka.testkit.AkkaSpec
-import akka.testkit.GHExcludeTest
 import akka.testkit.ImplicitSender
 import akka.testkit.SocketUtil.temporaryServerAddresses
 import akka.testkit.TestProbe
@@ -126,8 +125,7 @@ class UdpConnectedIntegrationSpec extends AkkaSpec("""
     }
 
     // #26903
-    // Excluded in GH Actions: https://github.com/akka/akka/issues/30462
-    "be able to send and receive when server goes away (and comes back)" taggedAs GHExcludeTest in {
+    "be able to send and receive when server goes away (and comes back)" in {
       val addresses = temporaryServerAddresses(2, udp = true)
       val serverAddress = addresses(0)
       val serverHandler = TestProbe()

--- a/akka-actor-tests/src/test/scala/akka/io/UdpConnectedIntegrationSpec.scala
+++ b/akka-actor-tests/src/test/scala/akka/io/UdpConnectedIntegrationSpec.scala
@@ -130,14 +130,17 @@ class UdpConnectedIntegrationSpec extends AkkaSpec("""
     "be able to send and receive when server goes away (and comes back)" taggedAs GHExcludeTest in {
       val addresses = temporaryServerAddresses(2, udp = true)
       val serverAddress = addresses(0)
-      val clientAddress = addresses(1)
-      val server = bindUdp(serverAddress, testActor)
-      val data1 = ByteString("To infinity and beyond!")
+      val serverHandler = TestProbe()
+      val server = bindUdp(serverAddress, serverHandler.ref)
 
-      val clientCommander = connectUdp(Some(clientAddress), serverAddress, testActor)
+      val clientAddress = addresses(1)
+      val clientHandler = TestProbe()
+      val clientCommander = connectUdp(Some(clientAddress), serverAddress, clientHandler.ref)
+
+      val data1 = ByteString("To infinity and beyond!")
       clientCommander ! UdpConnected.Send(data1)
 
-      expectMsg(Udp.Received(data1, clientAddress))
+      serverHandler.expectMsg(Udp.Received(data1, clientAddress))
 
       server ! Udp.Unbind
       expectMsg(Udp.Unbound)
@@ -152,11 +155,12 @@ class UdpConnectedIntegrationSpec extends AkkaSpec("""
       expectMsg(2)
 
       // when a new server appears at the same port it it should be able to receive
-      val serverIncarnation2 = bindUdp(serverAddress, testActor)
+      val serverIncarnation2Handler = TestProbe()
+      val serverIncarnation2 = bindUdp(serverAddress, serverIncarnation2Handler.ref)
       val dataToNewIncarnation = ByteString("Data to new incarnation")
       clientCommander ! UdpConnected.Send(dataToNewIncarnation, 3)
       expectMsg(3)
-      expectMsg(Udp.Received(dataToNewIncarnation, clientAddress))
+      serverIncarnation2Handler.expectMsg(Udp.Received(dataToNewIncarnation, clientAddress))
 
       serverIncarnation2 ! Udp.Unbind
     }

